### DISCUSSION TITLE
Disable the `prefer-destructuring` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#225] [General]
There is a proposal to disable the `prefer-destructuring` rule.

> `prefer-destructuring`: dot notation can be useful to determine where some value is coming from. It can also avoid variable name collisions. Also, forcing destructuring often introduces unnecessary verbosity.
> \- @edgardz 

Please see [the documentation](https://eslint.org/docs/latest/rules/prefer-destructuring)

**Comment by leroykorterink** [comment]
This rule made sense before [optional chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) existed, I don't think there are any good reasons to keep it disabled nowadays.

**Comment by ThaNarie** [comment]
I recently created an issue because `prefer-destructuring` conflicted with `no-unreadable-array-destructuring`:
https://github.com/mediamonks/eslint-config/issues/209

It was to only disable Arrays, but leave it for Objects.

---

Personally I do prefer destructuring in general, but recognise that there are scenarios where object-paths are preferable (especially when calling functions where scope should be maintained).

I'm slightly worried that less people will destructure where it's clearly favourable, resulting in PR feedback and code changes by others, but that shouldn't prevent disabling this rule.
